src/Pact/Analyze/Types/Languages.hs

```diff
@@ -346,9 +407,23 @@ pattern POr a b = CoreProp (Logical OrOp [a, b])
 pattern PNot :: Prop Bool -> Prop Bool
 pattern PNot a = CoreProp (Logical NotOp [a])
 
-mkDecimal :: Decimal.Decimal -> Decimal
-mkDecimal (Decimal.Decimal places mantissa) = fromRational $
-  mantissa % 10 ^ places
+decimalIso :: Iso' Decimal.Decimal Decimal
+decimalIso = iso mkDecimal unMkDecimal
+  where
+    mkDecimal :: Decimal.Decimal -> Decimal
+    mkDecimal (Decimal.Decimal places mantissa) = fromRational $
+      mantissa % 10 ^ places
+
+    unMkDecimal :: Decimal -> Decimal.Decimal
+    unMkDecimal algReal = case Decimal.eitherFromRational (toRational algReal) of
+      Left err  -> error err
+      Right dec -> dec
+
+fromPact :: Iso' a b -> a -> b
```

Comment:
perhaps these should work over `newtype`-wrapped `Iso`s? since it makes an assumption that for `Iso' a b`, that `a` corresponds to pact and `b` corresponds to analyze

Reply:
this is a good idea